java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java

```diff
@@ -3155,7 +3156,8 @@ public static boolean isCompatibleWithSwitchExpression(SwitchTree st) {
                             }
                             firstCase = false;
                         }
-                        if (leftTreeName != null && leftTreeName.contentEquals(getLeftTreeName(statements.get(0)))) {
+                        Name exprTree = getLeftTreeName(statements.get(0));
+                        if (leftTreeName != null && exprTree != null && leftTreeName.contentEquals(exprTree)) {
```

Comment:
I would introduce a method for checking conditions because it's the same as above.
cc: @Akshay-Gupta-Oracle